### ISSUE TITLE
quantization pact 

### ISSUE THREAD

**Issue by vnaveen0** [NervanaSystems/distiller#382]
#316 
If you take the stock example for PACT as mentioned in : `distiller/examples/quantization/quant_aware_train/preact_resnet20_cifar_pact.yaml`
and run it as it is with 

    logging level set to DEBUG in : logging.conf

 I get the following  message: 

    "No mask for parameter module.layer2.0.conv1.weight"
   
   This message comes from the following function in  scheduler.py 
    
	    def apply_mask(self, parameter):

for every parameter. I went and verified that  *"self.mask"* is never set, which can be seen when the model is saved

    
    class ParameterMasker(object):
    	self.mask = None
    
    called through: "apputils.save_checkpoint" in compress_classifier.py
    def state_dict(self):

2. I also went and checked: 

    class LearnedClippedLinearQuantizeSTE(torch.autograd.Function):
      * only forward method is called 
     * backward method is not called 

3. Can you please explain:
	3.1. If this is the default behaviour?
	3.2 If yes, then how is the model being quantized and how can I extract quantized weights?

**Comment by yiminchi** [comment]
@vnaveen0 
Have you extracted quantized weights successfully?

**Comment by nzmora** [comment]
Hi @vnaveen0 ,

Sorry for the late answer.  The log message that you see (`No mask for parameter module.layer2.0.conv1.weight`) is not an error and this is why it is only output when the log level is set to `DEBUG`.  I will remove this log, since we've never found it useful. Function `apply_mask` applies a pruning mask and, in this case, the log just confirms what you would expect: that there's no masking.

Cheers,
Neta

**Comment by vnaveen0** [comment]
@yiminchi  -  ".weights" are quantized weights and ".float_weights" are original floating point copy of the weights.
@nzmora Thanks for the response
